app/build.gradle

```diff
@@ -81,6 +81,16 @@ android {
         resValues true
         viewBinding true
     }
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+
```

Comment:
```suggestion
```

This is unnecessary, it has been covered by https://github.com/tuskyapp/Tusky/blob/05c7e7b8066506830f6c0bf9112e7beab90ef00c/build.gradle#L13-L17

Reply:
Check my comment above about why I disabled the toolchain. Hilt doesn't work with Java 21 bytecode and lowering the toolchain version forces you to build the project using an older Java version.

Reply:
The whole point of the toolchain is to be configured separately from the JDK used by Android Studio or the JDK running Gradle itself, so the project is always built using a specific version of the JDK no matter which JDK is installed on the machine. The Java 21 toolchain used to work on my machine, but not with Hilt.

The real question is: why do we want to use a toolchain in the first place, especially for Android projects?

Reply:
1. Hilt is compatible with Java 21, you have to set Java 21 as the `JAVA_HOME` or use it in your AS.
2. As it's convenient for configuring `targetCompatibility`, `jvmTarget`, and so on. At least, in the past. If we have to get rid of toolchain, it would be better to address the change in a separate PR.

<img width="825" alt="image" src="https://github.com/tuskyapp/Tusky/assets/10363352/e274a53e-c8b5-465f-88b9-eb5bf7bddd3b">

Reply:
For the record I also tried to configure the toolchain like this which is the recommended way for Kotlin projects:

```
kotlin {
    jvmToolchain(21)
}
```

But the effect is the same and Hilt doesn't seem to run inside the toolchain and will only recognize Java 21 sources when Gradle itself is run with JDK 21.

Reply:
Kotlin's `jvmToolchain` is honoring Java's, no need to declare it repeatedly.